src/controllers/confirm.company.controller.ts

```diff
@@ -17,10 +18,36 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
 };
 
 export const post = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const company: CompanyProfile = await getCompanyProfile(req.query.companyNumber as string);
+    if (isStatusEligible(company)) {
+      create(req);
+      const url = CONFIRMATION_STATEMENT_TRADING_STATUS.replace(":companyNumber", company.companyNumber);
+      return res.redirect(url);
+    } else {
+      return res.render(Templates.INVALID_STATUS);
+    }
+  } catch (e) {
+    next(e);
+  }
+};
+
+const isStatusEligible = (company: CompanyProfile): boolean => {
+  const status: string = company.companyStatus;
+  const invalidStatuses: string[] = INVALID_COMPANY_STATUSES.split(",");
+  let isValid: boolean = true;
+  invalidStatuses.forEach((invalid) => {
+    if (status === invalid) {
+      isValid = false;
+    }
+  });
+  return isValid;
+};
+
+const create = async (req: Request) => {
```

Comment:
can you rename the create function, it's not clear what it's creating when read in the function that calls it.

Reply:
In the function createConfirmationStatement is called so I just put create so as not to clash with that.